File: Android/App/app/src/main/java/com/efalco777/crud_products/presentation/product/viewmodel/products/ProductsViewModelState.kt
package com.efalco777.crud_products.presentation.product.viewmodel.products

import com.efalco777.crud_products.presentation.core.viewmodel.DataState
import com.efalco777.crud_products.presentation.product.model.PresentationProduct

data class ProductsViewModelState(
    val products: List<PresentationProduct>,
    val dataState: DataState
) {
    companion object {
        fun initial() =
            ProductsViewModelState(
                products = listOf(),
                dataState = DataState.Loading
            )
    }
}

File: iOS/App/ProductsCrud/Presentation/Product/ViewModel/Products/ProductsViewModelState.swift
//
//  ProductViewModelState.swift
//  ProductsCrud
//
//  Created by Emil Falkowski on 11/11/2024.
//

struct ProductsViewModelState {
    let dataState: DataState
    let products: [PresentationProduct]

    static func initial() -> ProductsViewModelState {
        return ProductsViewModelState(dataState: .loading, products: [])
    }
}
